### PR TITLE
Improve/update docs for `NodeKind`

### PR DESCRIPTION
**Pull request by kkysen**
It might be easier to review each commit individually so that it's easier to see the differences.

### CODE REVIEW

File: pdg/src/graph.rs [hunk 1]
```diff
@@ -15,47 +15,85 @@ use crate::util::ShortOption;
...
-    /// Pointer arithmetic.  The `isize` is the concrete offset distance.  We use this to detect
-    /// when two pointers always refer to different indices.
+
+    /// Pointer arithmetic.
+    ///
+    /// The [`isize`] is the concrete offset distance.
+    /// We use this to detect when two pointers always refer to different indices.
     Offset(isize),
 
     // Operations that can't have a `source`.
```
Comment (kkysen) [inline pdg/src/graph.rs:37]: What's this referring to?

File: pdg/src/graph.rs [hunk 2]
```diff
@@ -15,47 +15,85 @@ use crate::util::ShortOption;
...
+    /// Details TBD.
     IntToPtr,
-    /// The result of loading a value through some other pointer.  Details TBD.
+
+    /// The result of loading a value through some other pointer.
+    ///
+    /// Details TBD.
     LoadValue,
 
     // Operations that can't be the `source` of any other operation.
```
Comment (kkysen) [inline pdg/src/graph.rs:74]: What's this referring to?

File: pdg/src/graph.rs [hunk 3]
```diff
@@ -15,47 +15,85 @@ use crate::util::ShortOption;
 
 #[derive(Debug, Eq, PartialEq, Hash, Clone)]
 pub enum NodeKind {
-    /// A copy from one local to another.  This also covers casts such as `&mut T` to `&T` or `&T`
-    /// to `*const T` that don't change the type or value of the pointer.
+    /// A [`Copy`](Self::Copy) from one [`Local`] to another.
```
Comment (aneksteind) [inline pdg/src/graph.rs:18]: i think it's fine to refer to "copy" colloquially here, no need to self-reference

File: pdg/src/graph.rs (hunk 1 above)
Comment (aneksteind) [reply pdg/src/graph.rs:37]: it refers to `Node::source`, which will be `None` for all `Node`s with any of the following `NodeKind`s

File: pdg/src/graph.rs (hunk 2 above)
Comment (aneksteind) [reply pdg/src/graph.rs:74]: Like above, refers to `Node::source`, which can never point to a `Node` (via `Node::dest`) which has any of the following `NodeKind`s

File: pdg/src/graph.rs (hunk 1 above)
Comment (kkysen) [reply pdg/src/graph.rs:37]: I meant what is the comment and operation referring to.  Which variant?

**Review by aneksteind — APPROVED** [review]
I imagine you'll want to update the documentation in response to my comments and I trust what you put will be sufficient, so LGTM

File: pdg/src/graph.rs (hunk 1 above)
Comment (aneksteind) [reply pdg/src/graph.rs:37]: all of them that follow until the next similar comment

File: pdg/src/graph.rs (hunk 3 above)
Comment (kkysen) [reply pdg/src/graph.rs:18]: Okay that makes sense.  I'll change that back.

File: pdg/src/graph.rs (hunk 1 above)
Comment (kkysen) [reply pdg/src/graph.rs:37]: Ahh, got it.  I'll make that clearer.

Comment (kkysen) [reply pdg/src/graph.rs:37]: I'll put that in the docs for each variant it applies to, as I think doing it as is is a bit brittle if the variants are ever moved around, and it wasn't that clear to begin with.

File: pdg/src/graph.rs (hunk 3 above)
Comment (kkysen) [reply pdg/src/graph.rs]: Fixed in https://github.com/immunant/c2rust/pull/658/commits/40f6ecfd73752c804da9be32c67d79acc36dcade.

File: pdg/src/graph.rs (hunk 1 above)
Comment (kkysen) [reply pdg/src/graph.rs]: Fixed in https://github.com/immunant/c2rust/pull/658/commits/dc29cfaabeb0dbc8764247e84decdad82ad3f3e8.

File: pdg/src/graph.rs (hunk 2 above)
Comment (kkysen) [reply pdg/src/graph.rs]: Fixed in https://github.com/immunant/c2rust/pull/658/commits/dc29cfaabeb0dbc8764247e84decdad82ad3f3e8.

**Pull request merged**